### ISSUE TITLE
Direct Charge on Platform Account Customer to Connected Account Not working

### ISSUE THREAD

**Issue by anshul35** [stripe/stripe-go#499]
Stripe-go version : 2568e0f2a1ab55fa486a10e8d6797d2dce0c70b5

**Scenario**

Platform Account: A
Connected Account: B
Customer Registered to A: X

**Expectation**:

Charge Customer X on his default source and and put the money in account B.

**Error** 

```
{\"status\":400,\"message\":\"You provided a customer without specifying a source. The default source of the customer is a source and cannot be shared from existing customers.\\n\",\"request_id\":\"req_rqGRipNZaiDJbM\",\"type\":\"invalid_request_error\"}
```

**Code**
```
**CreateToken**:
        stripe.Key = config.STRIPE_CLIENT_SECRET
	params := &stripe.TokenParams{
		Customer: stripeCustomerId,
	}
	params.SetStripeAccount(connectAccountId)
	return token.New(params)

**CreateChargeParams**:
      params := stripe.ChargeParams{
		Currency: "usd",		
		Amount: uint64(userBill.Amount * 100), //Convert the amount in cents, lowest unit of USD
		Fee: 0, 
	}
	params.SetStripeAccount(merchantStripeInfo.AccountID)

	//Set source
	token, err := CreateToken(stripeCustomer.ID, connectedAccountId)
	if err != nil {
		return nil, err
	}
	params.SetSource(token.ID)

**Charge**:
       chargeParams, err := CreateChargeParams(userBill, stripeCustomer)
	if err != nil {
		logger.Errorf("Unable to create charge params for userbill for customer ID = %s with error = %s",
			stripeCustomer.ID, err.Error())
		return err
	}
	//Try with the default source
	ch, err := charge.New(chargeParams)
```

**Comment by ob-stripe** [comment]
Hi @anshul35, thanks for reporting this!

In order to be able to share a card source, the API requires that the source ID be explicitly passed in the `card` parameter, even if the card source is the customer's default source.

I think the Go client library does not support specifying this parameter, but I could be mistaken. cc @brandur-stripe @remi-stripe, can one of you confirm this is the case?

**Comment by remi-stripe** [comment]
So there are 2 issues at hand here. First, we don't support passing `card="card_XXXX"` when creating a Token which is what @ob-stripe is talking about. Second, we don't support cloning a Source which is the API described [here](https://stripe.com/docs/sources/connect#shared-card-sources) and what @anshul35 needs right now.

Going to open a PR in a few minutes.

**Comment by remi-stripe** [comment]
PR up in https://github.com/stripe/stripe-go/pull/500

**Comment by brandur-stripe** [comment]
Released the patch in #500 as 28.8.0.

**Comment by anshul35** [comment]
Thanks @remi-stripe @brandur-stripe @ob-stripe  for a quick turn out. Will try the fix out.
